kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -74,44 +84,67 @@ export const ExpressionContainer: (props: ExpressionContainerProps) => JSX.Eleme
         ]}
       />
     );
-  };
+  }, [i18n.clear, actionDropdownIsOpen, logicTypeIsPresent, executeClearAction]);
+
+  const getLogicTypesWithoutUndefined = useCallback(() => {
+    return Object.values(LogicType).filter((logicType) => logicType !== LogicType.Undefined);
+  }, []);
+
+  const renderLogicTypeItems = useCallback(() => {
+    return _.map(getLogicTypesWithoutUndefined(), (key) => <SimpleListItem key={key}>{key}</SimpleListItem>);
+  }, [getLogicTypesWithoutUndefined]);
 
-  const buildLogicSelectorMenu = () => {
+  const buildLogicSelectorMenu = useCallback(() => {
     return (
       <PopoverMenu
         title={i18n.selectLogicType}
-        arrowPlacement={() => document.getElementById("expression-container-box")!}
+        arrowPlacement={() => document.querySelector(".expression-container-box")! as HTMLElement}
         body={<SimpleList onSelect={onLogicTypeSelect}>{renderLogicTypeItems()}</SimpleList>}
       />
     );
-  };
-
-  const renderLogicTypeItems = () => {
-    return _.map(
-      [
-        i18n.literalExpression,
-        i18n.context,
-        i18n.decisionTable,
-        i18n.relation,
-        i18n.function,
-        i18n.invocation,
-        i18n.list,
-      ],
-      (key) => <SimpleListItem key={key}>{key}</SimpleListItem>
-    );
-  };
+  }, [i18n.selectLogicType, onLogicTypeSelect, renderLogicTypeItems]);
```

Comment:
`i18n.selectLogicType` should be constant, and therefore, doesn't need to be a depenency.

Reply:
if I remove such constant from dependencies, eslint complains:
![image](https://user-images.githubusercontent.com/22591802/100775627-d2bbc780-3403-11eb-9748-7cdece1d3f50.png)

Reply:
Hmm..... that's interesting :)

In fact, ESLint is correct. `i18n.selectLogicType` should be constant IF the locale doesn't change. When the locale changes, we have to re-render this component. So please ignore my original comment and keep trusting ESLint :D 

Btw, since this is a callback with no parameters, maybe this can be converted into a memo? :)